### ISSUE TITLE
[Refactor] Rewrite the settings tab in React

### ISSUE THREAD

**Issue by aptend** [aptend/typing-transformer-obsidian#59]
working on the branch [react](https://github.com/aptend/typing-transformer-obsidian/tree/react)

**Comment by caasion** [comment]
Are you sure you want to use React? It seems like svelte is a more popular option now, but if you are familiar with React, this would be good.

**Comment by aptend** [comment]
Nope, I'm not familiar with any frontend frameworks 😂 . It is good to know svelte! I will take a look.

**Comment by aptend** [comment]
I guess the Svelte is a better option. It has no runtime and is suitable for small projects like the settings tab.

**Comment by caasion** [comment]
Svelte is definitely a lot cleaner in terms of the code. I have some experience with svelte (I have been trying to learn it), so I hope I will be able to help with that.

**Comment by caasion** [comment]
Also, what is the scope that you want to refactor? Do you want to refactor the editor + profiles or the whole settings tab including the "auto format", "zone indicator", "rules" etc.?

I was thinking to only refactor the editor because it is more complex, but if you want a different view for the other settings, it would be a good idea to refactor those too.

**Comment by aptend** [comment]
Yes, only refactoring the editor is enough. I am planning to reuse the editor component to edit the rules for space-adding in the future. 

btw, recently I am quite busy, sorry can't help with coding much. Really really thank you for your efforts 👍

**Comment by caasion** [comment]
I've also been quite busy and I haven't been home that much. I've been trying to figure out how to compile this plugin (and any other plugin) on windows but I'm not sure how. If you have any resources on this, please send.

This refactor might also be quite slow because I am trying to figure out code mirror (I don't have experience with how it works.) so I will definitely need your help in some aspects.

Btw, do you have any contacts or communications such as discord?

**Comment by aptend** [comment]
https://github.com/aptend/typing-transformer-obsidian/blob/main/docs/develop.md 
This is how I compile this plugin on windows. which step do you have trouble at?

never used discord before, but I will try it. Maybe you can let me how to contact you in discord, my email is aptend@hotmail.com

**Comment by caasion** [comment]
> never used discord before, but I will try it. Maybe you can let me how to contact you in discord, my email is [aptend@hotmail.com](mailto:aptend@hotmail.com)

My discord username/tag is `caasi#3126`

You should also join OMG (Obsidian official server): https://obsidian.md/community
